components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/test/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/DefaultRequestCoordinatorTest.java

```diff
@@ -185,4 +203,78 @@ public void testCookieValidationFailedException() {
+            when(request.getParameter(FrameworkConstants.RequestParams.ISSUER)).thenReturn(relyingParty);
+            when(request.getParameter(TENANT_DOMAIN)).thenReturn(tenantDomain);
+            when(context.getRequestType()).thenReturn(requestType);
+            when(context.getTenantDomain()).thenReturn(tenantDomain);
+            when(context.getServiceProviderName()).thenReturn("consoleApplication");
+            when(context.getEndpointParams()).thenReturn(new HashMap<>());
+
+            // Mocking FrameworkUtils
+            frameworkUtils.when(() -> FrameworkUtils.getContextData(request)).thenReturn(context);
+
+            // Mocking ApplicationManagementService behavior
+            ApplicationManagementServiceImpl mockApplicationManagementService =
+                    mock(ApplicationManagementServiceImpl.class);
+            applicationManagementService.when(ApplicationManagementService::getInstance)
+                    .thenReturn(mockApplicationManagementService);
+
+            // Mocking ServiceProvider and its properties
+            ServiceProvider serviceProvider = mock(ServiceProvider.class);
+            when(serviceProvider.isApplicationEnabled()).thenReturn(false);  // ServiceProvider is disabled
+            when(mockApplicationManagementService.getServiceProviderByClientId(anyString(), anyString(), anyString()))
+                    .thenReturn(serviceProvider);
+
+            ConfigurationFacade configurationFacadeInstance = mock(ConfigurationFacade.class);
+            configurationFacade.when(ConfigurationFacade::getInstance).thenReturn(configurationFacadeInstance);
+            when(configurationFacadeInstance.getAuthenticationEndpointRetryURL())
+                    .thenReturn("https://localhost:9443/retry");
+
+            frameworkUtils.when(() -> FrameworkUtils.sendToRetryPage(any(), any(), any(), any(), any())).
+                    thenCallRealMethod();
+
+            frameworkUtils.when(() -> FrameworkUtils.getRedirectURL(any(), any())).thenCallRealMethod();
+
+            frameworkUtils.when(() -> FrameworkUtils.addAuthenticationErrorToCache(any(), any(), any())).
+                    thenAnswer(invocation -> null);
+
+            String expectedRedirectURL = new URIBuilder("https://localhost:9443/retry")
+                    .addParameter("status", ERROR_STATUS_APP_DISABLED)
+                    .addParameter("statusMsg", ERROR_DESCRIPTION_APP_DISABLED)
+                    .build().toString();
+
+            frameworkUtils.when(() -> FrameworkUtils.getRedirectURLWithFilteredParams(any(),
+                            (AuthenticationContext) any())).thenCallRealMethod();
+
+            frameworkUtils.when(() -> FrameworkUtils.getRedirectURLWithFilteredParams(any(),
+                            (Map<String, Serializable>) any())).thenCallRealMethod();
+
+            defaultRequestCoordinator.handle(request, response);
+
+            Assert.assertEquals(expectedRedirectURL, response.getRedirectURL());
```

Comment:
instead of assert equals lets assert whether this is a redirect to the retry page and also pull the status and statusMsg from the url and asset against the messages that should be there. 

the problem with assert equals is if there are any other additional params comming in the redirect from framework this will fail.